File: problems/Java/38. Count and Say.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        
        String res = "1";
        for (int i = 2; i <= n; i++) {
            StringBuilder next = new StringBuilder();
            int j = 0;
            while (j < res.length()) {
                int cnt = 1;
                while (j + 1 < res.length() && res.charAt(j) == res.charAt(j + 1)) {
                    cnt++;
                    j++;
                }
                next.append(cnt);
                next.append(res.charAt(j));
                j++;
            }
            res = next.toString();
        }
        return res;
    }
}

File: problems/Python/38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        
        res = "1"
        for _ in range(n - 1):
            next_res = ""
            i = 0
            while i < len(res):
                cnt = 1
                while i + 1 < len(res) and res[i] == res[i + 1]:
                    cnt += 1
                    i += 1
                next_res += str(cnt) + res[i]
                i += 1
            res = next_res
            
        return res